lib/schmobile/middleware.rb

```diff
@@ -31,7 +31,7 @@ def redirect?(request)
         redirecting = false
       end
 
-      redirecting ? redirect_location(request) : nil
```

Comment:
It is. The method ends with `?` so it should return true or false, not a string or nil.
Also the only place where this method is used here is `if redirect? then redirect to: redirect_location(request)`